hazelcast/src/main/java/com/hazelcast/client/impl/connection/tcp/TcpClientConnectionManager.java

```diff
@@ -462,6 +467,15 @@ Connection connect(Object target, Function<Object, Connection> getOrConnectFunct
         } catch (ClientNotAllowedInClusterException e) {
             logger.warning("Exception during initial connection to " + target + ": " + e);
             throw e;
+        } catch (TargetDisconnectedException e) {
```

Comment:
When can the connect methods throw `TargetDisconnectedException`? As far as I see, we can only throw this exception after the connection is established